.github/workflows/deploy-portainer.yml

```
@@ -1,539 +1,578 @@
-          exit 1
+                  if [ -f stack.env ]; then
+                    docker compose -p "$COMPOSE_PROJECT_NAME" --env-file stack.env build backend
+                    BACKEND_DEPLOY_VERSION="$TARGET_VERSION" docker compose -p "$COMPOSE_PROJECT_NAME" --env-file stack.env up -d backend redis
+                  elif [ -f .env ]; then
+                    docker compose -p "$COMPOSE_PROJECT_NAME" --env-file .env build backend
+                    BACKEND_DEPLOY_VERSION="$TARGET_VERSION" docker compose -p "$COMPOSE_PROJECT_NAME" --env-file .env up -d backend redis
+                  else
+                    docker compose -p "$COMPOSE_PROJECT_NAME" build backend
+                    BACKEND_DEPLOY_VERSION="$TARGET_VERSION" docker compose -p "$COMPOSE_PROJECT_NAME" up -d backend redis
+                  fi
+EOF
```

actionlint (1.7.12)

[error] could not parse as YAML: could not find expected ':'

(syntax-check)